src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/MSBuildProjectManager.cs

```diff
@@ -93,49 +92,151 @@ public async void ProjectLoaded(ProjectLoadedEventArgs args)
         {
             try
             {
-                HandleDebug(args.ProjectInstance);
+                await ProjectLoadedAsync(args);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Unexpected exception got thrown from the Razor plugin: " + ex);
+            }
+        }
 
-                if (!TryResolveConfigurationOutputPath(args.ProjectInstance, out var configPath))
-                {
-                    return;
-                }
+        public void RazorDocumentChanged(RazorFileChangeEventArgs args)
```

Comment:
Why is this not async?

Reply:
It's tied to file system watcher events which dispatch synchronously. Also async void for events is a supported pattern if we need to do async work